src/oracle/error.cairo

```diff
@@ -1,3 +1,173 @@
 mod OracleError {
+    use starknet::ContractAddress;
+
     const ALREADY_INITIALIZED: felt252 = 'already_initialized';
+
+    fn NON_EMPTY_TOKENS_WITH_PRICES(data_1: u32) {
+        let mut data = array!['non empty tokens prices'];
+        data.append(data_1.into());
+        panic(data)
+    }
+
+    fn DUPLICATED_TOKEN_PRICE() {
+        let mut data = array![];
+        data.append('duplicated token price');
+        panic(data)
+    }
+
+    fn EMPTY_PRIMARY_PRICE() {
+        let mut data = array![];
+        data.append('empty primary price');
+        panic(data)
+    }
+
+    fn EMPTY_PRICE_FEED_MULTIPLIER() {
+        let mut data = array![];
+        data.append('empty price feed multiplier');
+        panic(data)
+    }
+
+    fn INVALID_MIN_MAX_BLOCK_NUMBER(data_1: u64, data_2: u64) {
+        let mut data = array!['invalid min max block number'];
+        data.append(data_1.into());
+        data.append(data_2.into());
+        panic(data)
```

Comment:
why not do this?
```suggestion
        panic(array!['invalid min max block number', data_1.into(), data_2.into()])
```

Reply:
Agreed, the code will benefit from enhanced readability. I'll work on implementing the changes.